tests/agents/handagent/conftest.py

```diff
@@ -48,6 +48,6 @@ def env(request):
                                                                show_viewport=False,
                                                                uuid=str(uuid.uuid4()))
 
-    shared_env.reset()
-
-    return shared_env
+    with shared_env:
+        shared_env.reset()
+        yield shared_env
```

Comment:
I can't run the Holodeck tests on my laptop but I had a chance to do some experimentation with pytest locally. What we want is to reset every time but keep the same environment, and this will actually dispose the environment after finishing every test function. A good way to get around that is to pull the environment init and cleanup into another fixture with `@pytest.fixture(scope="module")` (maybe call it `shared_env` to replace the global) that is called once for all tests in the directory and then create another with just `@pytest.fixture` that depends on `shared_env` to reset for every test.

Something like:

```python
@pytest.fixture(scope="module")
def shared_env(request):
    cfg = request.param
    binary_path = holodeck.packagemanager.get_binary_path_for_package("DefaultWorlds")

    with holodeck.environments.HolodeckEnvironment(scenario=cfg,
                                                   binary_path=binary_path,
                                                   show_viewport=False,
                                                   uuid=str(uuid.uuid4())) as env:
    # resetting is handled by `env` and cleanup happens immediately after exiting the module
    yield env


@pytest.fixture
def env(request, shared_env):
    shared_env.reset()
    return shared_env
```